agent/consul/acl.go

```diff
@@ -1438,6 +1438,30 @@ func (f *aclFilter) filterNodeDump(dump *structs.NodeDump) {
 	*dump = nd
 }
 
+// filterServiceDump is used to filter nodes based on ACL rules.
+func (f *aclFilter) filterServiceDump(services *structs.ServiceDump) {
+	svcs := *services
+	var authzContext acl.AuthorizerContext
+
+	for i := 0; i < len(svcs); i++ {
+		service := svcs[i]
+		service.Service.FillAuthzContext(&authzContext)
```

Comment:
`service.Service` can be `nil` here. We could potentially just rely on `service.GatewayService` here for ACLs, since the RPC output is all based off of that output.